tests/taskmodules/test_re_text_classification_with_indices.py

```diff
@@ -2882,3 +2964,86 @@ def test_encode_with_add_entity_tags_to_input_windowing(documents, insert_marker
                 ("[SEP]", "O"),
             ],
         ]
+
+
+def test_create_annotations_from_output(taskmodule, documents):
+    document = documents[4]
+    model_output = {
+        "labels": torch.tensor([2, 3, 1]),
+        "probabilities": torch.tensor(
+            [
+                # O, org:founded_by, per:employee_of, per:founder
+                [0.1, 0.1, 0.6, 0.2],
+                [0.2, 0.1, 0.2, 0.5],
+                [0.1, 0.6, 0.1, 0.2],
+            ]
+        ),
+    }
+    task_encodings = taskmodule.encode(document, encode_target=True)
+    unbatched_model_outputs = taskmodule.unbatch_output(model_output)
+    result = []
+    for i in range(len(unbatched_model_outputs)):
+        result.extend(
+            list(
+                taskmodule.create_annotations_from_output(
+                    task_encoding=task_encodings[i], task_output=unbatched_model_outputs[i]
+                )
+            )
+        )
+    # assert result == list(document.relations)
+    scores = [0.6000000238418579, 0.5, 0.6000000238418579]
+    for i, ((layer_name, predicted_relation), original_relation) in enumerate(
+        zip(result, list(document.relations))
+    ):
+        assert layer_name == taskmodule.relation_annotation
+        assert predicted_relation == original_relation.copy()
+        assert predicted_relation.score == scores[i]
+
+
+def test_create_annotations_from_output_with_argument_and_relation_type_whitelist(
+    documents, taskmodule
+):
+    document = documents[4]
+    model_output = {
+        "labels": torch.tensor([2, 3, 1]),
+        "probabilities": torch.tensor(
+            [
+                # O, org:founded_by, per:employee_of, per:founder
+                [0.1, 0.1, 0.6, 0.2],
+                [0.2, 0.1, 0.2, 0.5],
+                [0.1, 0.6, 0.1, 0.2],
+            ]
+        ),
+    }
+    task_encodings = taskmodule.encode(document, encode_target=True)
```

Comment:
It is really hard to understand, what's happening here. On idea for improvement would be to show the candidate relations that will get classified:
```suggestion
    task_encodings = taskmodule.encode(document, encode_target=True)
    candidate_relations = [
        task_encoding.metadata["candidate_annotation"] for task_encoding in task_encodings
    ]
    resolved_candidate_relations = [rel.resolve() for rel in candidate_relations]
    assert resolved_candidate_relations == [
        ("per:employee_of", (("PER", "Entity G"), ("ORG", "H"))),
        ("per:founder", (("PER", "Entity G"), ("ORG", "I"))),
        ("org:founded_by", (("ORG", "I"), ("ORG", "H"))),
    ]
```